test/unit/user_test.rb

```diff
@@ -12,31 +12,64 @@ class UserTest < ActiveSupport::TestCase
     test 'user model cannot be created without first name' do
       @user.first_name = ""
       @user.save
-      assert_equal(@user.errors.messages[:first_name],["can't be blank", "is invalid"])
+      assert_equal(["can't be blank", "is invalid"], @user.errors.messages[:first_name])
     end
   end
 
   [generate_email].each do |new_email|
     test 'user model cannot be created without last name' do
       @user.last_name = ""
       @user.save
-      assert_equal(@user.errors.messages[:last_name],["can't be blank", "is invalid"])
+      assert_equal(["can't be blank", "is invalid"], @user.errors.messages[:last_name])
     end
   end
 
   [generate_email].each do |new_email|
     test 'user model cannot be created without pin' do
       @user.pin = ""
       @user.save
-      assert_equal(@user.errors.messages[:pin],["can't be blank", "requires numbers only.", "must be 4 digits."])
+      assert_equal(["can't be blank", "requires numbers only.", "must be 4 digits."], @user.errors.messages[:pin])
+    end
+  end
+
+  [generate_email].each do |new_email|
+    test 'user model cannot be created with an existing alternate e-mail address in database' do
+      @user.save
+      user_two = crank!(:user, alt_email: @user.alt_email)
+      user_two.save
+      assert_equal(["has already been taken"], user_two.errors.messages[:alt_email])
+    end
+  end
+
+  [generate_email].each do |new_email|
+    test 'user model cannot be created with 4 of the same repeated digits as pin' do
+      @user.pin = "1111"
+      @user.save
+      assert_equal(@user.errors.messages[:pin],["does not meet our requirements. Please try again."])
+    end
+  end
+
+  [generate_email].each do |new_email|
+    test 'user model cannot be created with alternate repeated digits as pin' do
+      @user.pin = "1212"
+      @user.save
+      assert_equal(@user.errors.messages[:pin],["does not meet our requirements. Please try again."])
+    end
+  end
+
+  [generate_email].each do |new_email|
+    test 'user model cannot be created with 3 of the same repeated digits in a row as pin' do
+      @user.pin = "0007"
+      @user.save
+      assert_equal(@user.errors.messages[:pin],["does not meet our requirements. Please try again."])
```

Comment:
But I think we changed the error msg wording, right?

Reply:
We changed it this per Courtney's approval.